### ISSUE TITLE
Dirty directory error, but cargo-workspaces is responsible

### ISSUE THREAD

**Issue by aasseman** [pksunkara/cargo-workspaces#103]
I am trying to use `cargo workspaces publish --from-git` on [my repo](https://github.com/semiotic-ai/timeline-aggregation-protocol), but I get:
```
error: 1 files in the working directory contain changes that were not yet committed into git:

Cargo.toml

to proceed despite this and include the uncommitted changes, pass the `--allow-dirty` flag
error: unable to publish package tap_core
```
The problem is that I am 100% sure the directory is not dirty, but in fact `cargo workspaces publish` modifies the `Cargo.toml`, then complains about it. Here's a video:

https://github.com/pksunkara/cargo-workspaces/assets/33075224/52f4372a-0591-4ee3-afd6-76396f1424cc

**Comment by aasseman** [comment]
A workaround is to use `--no-remove-dev-deps`.

**Comment by pksunkara** [comment]
Yes, unfortunately, we had to enable this by default because of how cargo treats workspace dev-dependencies.

Maybe, there's a way to have a better smart default in the future.

**Comment by pksunkara** [comment]
Published 0.2.44 with a smarter default that should fix your issue.
